command/eval_list.go

```diff
@@ -0,0 +1,182 @@
+			"-verbose":    complete.PredictNothing,
+			"-job":        complete.PredictAnything,
+			"-status":     complete.PredictAnything,
+			"-per_page":   complete.PredictAnything,
+			"-page_token": complete.PredictAnything,
+		})
+}
+
+func (c *EvalListCommand) AutocompleteArgs() complete.Predictor {
+	return complete.PredictFunc(func(a complete.Args) []string {
+		client, err := c.Meta.Client()
+		if err != nil {
+			return nil
+		}
+
+		if err != nil {
+			return nil
+		}
+
+		resp, _, err := client.Search().PrefixSearch(a.Last, contexts.Evals, nil)
+		if err != nil {
+			return []string{}
+		}
+		return resp.Matches[contexts.Evals]
+	})
+}
+
+func (c *EvalListCommand) Name() string { return "eval status" }
+
+func (c *EvalListCommand) Run(args []string) int {
+	var monitor, verbose, json bool
+	var perPage int
+	var tmpl, pageToken, filterJobID, filterStatus string
+
+	flags := c.Meta.FlagSet(c.Name(), FlagSetClient)
+	flags.Usage = func() { c.Ui.Output(c.Help()) }
+	flags.BoolVar(&monitor, "monitor", false, "")
+	flags.BoolVar(&verbose, "verbose", false, "")
+	flags.BoolVar(&json, "json", false, "")
+	flags.StringVar(&tmpl, "t", "", "")
+	flags.IntVar(&perPage, "per_page", 0, "")
+	flags.StringVar(&pageToken, "page_token", "", "")
+	flags.StringVar(&filterJobID, "job", "", "")
+	flags.StringVar(&filterStatus, "status", "", "")
+
+	if err := flags.Parse(args); err != nil {
+		return 1
+	}
+
+	client, err := c.Meta.Client()
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error initializing client: %s", err))
+		return 1
+	}
+
+	opts := &api.QueryOptions{
+		PerPage:   int32(perPage),
+		NextToken: pageToken,
+		Params:    map[string]string{},
+	}
+	if filterJobID != "" {
+		opts.Params["job"] = filterJobID
+	}
+	if filterStatus != "" {
+		opts.Params["status"] = filterStatus
+	}
+
+	evals, qm, err := client.Evaluations().List(opts)
+	if err != nil {
+		c.Ui.Error(fmt.Sprintf("Error querying evaluations: %v", err))
+		return 1
+	}
+
+	// If args not specified but output format is specified, format
+	// and output the evaluations data list
+	if json || len(tmpl) > 0 {
+		out, err := Format(json, tmpl, evals)
+		if err != nil {
+			c.Ui.Error(err.Error())
+			return 1
+		}
+
+		c.Ui.Output(out)
+		return 0
+	}
+
+	if len(evals) == 0 {
+		c.Ui.Output("No evals found")
+		return 0
+	}
+
+	// Truncate the id unless full length is requested
+	length := shortId
+	if verbose {
+		length = fullId
+	}
+
+	out := make([]string, len(evals)+1)
+	out[0] = "ID|Priority|Triggered By|Job ID|Status|Placement Failures"
+	for i, eval := range evals {
+		failures, _ := evalFailureStatus(eval)
+		out[i+1] = fmt.Sprintf("%s|%d|%s|%s|%s|%s",
+			limit(eval.ID, length),
+			eval.Priority,
+			eval.TriggeredBy,
+			eval.JobID,
+			eval.Status,
+			failures,
+		)
+	}
+	c.Ui.Output(formatList(out))
+
+	if qm.NextToken != "" {
+		c.Ui.Output(fmt.Sprintf(`
+Results have been paginated. To get the next page run:
+
+nomad eval list -page_token %s`, qm.NextToken))
```

Comment:
I looked at doing that and then got lazy because `nomad plan` doesn't do it except for the `-var` and `-var-file` args. But you're right, we should try to do better here.

Reply:
Done!